### DISCUSSION TITLE
Filtering by a type (type is not in db, model is in db) field using 'not' doesn't work

### DISCUSSION THREAD

**Discussion by Redskinsjo** [prisma/prisma#19630] [Q&A]
## Problem

I have this type
``` 
type Details {
  birthday String?
  phone    String?
  email    String?
}
```

and this model
```
model profiles {
  id           String            @id @default(auto()) @map("_id") @db.ObjectId
  schools      School[]
  details      Details?
  skills       Skill[]
  allSkills    String?
}
```

I try to query filtering with a 'not'
```
const profiles = await prisma.profiles.findMany({
        where: {
          user: {
            email: {
              not: null,
            },
          },
        },
      });
```

but it's not working, typescript doesn't suggest 'not' for email (which is string | undefined | null ) whereas when I change prisma type to a prisma model like this: then typescript suggest 'not'
```
model Details {
  birthday String?
  phone    String?
  email    String?
}
```

SO WE CAN'T FILTER THE SAME WAY ON A TYPE THAN ON A MODEL ?

## Suggested solution

Filtering with 'not' should work for any type or model i guess

**Comment by janpio** [comment]
Your query uses `user`, but both your composite type and model do not include a field of that name. Is that correct?

**Reply by nurul3101** [reply]
@Redskinsjo Did you had a chance to look at Jan's response? Please let us know in case you still need any help. If not, please consider marking the discussion as answered/closed.

**Reply by Redskinsjo** [reply]
Thank you. I’m closing the issue.
